msyncd/synchronizer.cpp

```diff
@@ -2044,13 +2045,14 @@ void Synchronizer::removeExternalSyncStatus(const SyncProfile *aProfile)
 {
     int accountId = aProfile->key(KEY_ACCOUNT_ID).toInt();
     if (accountId) {
-        if (iExternalSyncProfileStatus.contains(accountId)) {
+        const QString &profileName = aProfile->name();
+        if (iExternalSyncProfileStatus.contains(profileName)) {
             // if profile was set to sync externally emit the change state signal
-            if (iExternalSyncProfileStatus.value(accountId)) {
+            if (iExternalSyncProfileStatus.value(profileName)) {
                 emit syncedExternallyStatus(accountId, aProfile->clientProfile()->name(),false);
```

Comment:
Profile name can be anything, in jolla we use subprofilename-accountId, but there are not guarantees, client profile on the other hand should match a plugin, so any listeners of this signals can rely on that since those should not change.